spec/log-window-tests.js

```diff
@@ -54,59 +67,79 @@ describe('Log window', () => {
   describe('Window handling', () => {
     it('opens the logging window', () => {
       EmeLogWindow.instance.open();
-      expect(window.open).toHaveBeenCalledWith(
-        'log.html', jasmine.any(String), jasmine.any(String));
+      expect(chrome.windows.create).toHaveBeenCalledWith(new Object({
+        url: 'log.html', type: 'popup', height: 600, width: 700
+      }));
     });
 
-    it('reports the logging window is open', () => {
-      mockWindow.closed = false;
-      EmeLogWindow.instance.open();
+    it('reports the logging window is open', async function (done) {
+      expect(EmeLogWindow.instance.isOpen()).toBe(false);
+      await EmeLogWindow.instance.open();
       expect(EmeLogWindow.instance.isOpen()).toBe(true);
+      done();
     });
 
-    it('reports the logging window is closed', () => {
-      mockWindow.closed = true;
-      EmeLogWindow.instance.open();
+    it('reports the logging window is closed', async function (done) {
+      await EmeLogWindow.instance.open();
+      expect(EmeLogWindow.instance.isOpen()).toBe(true);
+      EmeLogWindow.instance.close(mockWindow.id);
       expect(EmeLogWindow.instance.isOpen()).toBe(false);
+      done();
     });
   });
 
+  it('appends a text log', async function (done) {
+    await EmeLogWindow.instance.open();
+    EmeLogWindow.instance.appendLog('text log 1');
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual('text log 1');
+    done();
+  });
+
+  it('clears the text log', async function (done) {
+    await EmeLogWindow.instance.open();
+    EmeLogWindow.instance.appendLog('text log 2');
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual(jasmine.any(String));
+    EmeLogWindow.instance.clear();
+    expect(EmeLogWindow.instance.getTextLogs()).toEqual('');
+    done();
+  });
+
   it('logs with timestamps', () => {
     const date = new Date('July 20, 1969 12:34:56 UTC');
     EmeLogWindow.instance.open();
-    EmeLogWindow.instance.appendLog({
+    EmeLogWindow.instance.appendLog(appendLog({
```

Comment:
What is this extra layer of appendLog() inside of instance.appendLog()?  I can't find that defined anywhere.  Seems to be referenced in many places.

Reply:
`log.js::appendLog()` is being called here. I included log.js file to the `srcFiles` section in `jasmine-browser.json` file, too. Because, I can't test `appendLog()` API that has been moved from `log-window.js::EmeLogWindow.appendLog()` to `log.js::appendLog()` (this method gets called by `chrome.runtime.onMessage` in the log.js).

I was trying to mimic what the message handler is doing. Please advise me if there is a better way to do this. 

log.js:
```
  chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
    const textLog = appendLog(request.log);
    chrome.runtime.sendMessage({ type: 'EME_LOGGER_APPEND_LOG', data: textLog });
  });
```

log-window.js:
```
  chrome.runtime.onMessage.addListener((message, sender, sendResponse) => {
    ...
    } else if (message.type == 'EME_LOGGER_APPEND_LOG') {
      EmeLogWindow.instance.appendLog(message.data);
    }
  });
```